lexer.c

```diff
@@ -0,0 +1,75 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   lexer.c                                            :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: pcharton <pcharton@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2021/04/03 20:31:36 by pcharton          #+#    #+#             */
+/*   Updated: 2021/04/05 15:04:56 by pcharton         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "header.h"
+#include <stdbool.h>
+
+void	get_command(void)
+{
+	char	*line;
+	int		ret;
+
+	ret = get_next_line(0, &line);
+	if (ret == -1)
+	{
+		puts("gnl error");
+		exit(-1);
+	}
```

Comment:
There is a potential memory leak here.
GNL's line remainder could (And will) still be malloc'ed

Reply:
True. I did not plan any error managment yet. Will take care of.